test/Test/Anemone/Roundtrip.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Test.Anemone.Roundtrip where
+
+import           Anemone.Parser
+import           Anemone.Pretty
+
+import           Control.Monad.ST (runST, ST)
+
+import           Data.Array.ST (MArray, STUArray, newArray, readArray)
+import           Data.Array.Unsafe (castSTUArray)
+
+import           Disorder.Core.Run (ExpectedTestSpeed(..), disorderCheckEnvAll)
+import           Disorder.Jack (Property)
+import           Disorder.Jack (gamble, property, arbitrary, counterexample)
+
+import           P
+
+import           System.IO (IO)
+
+
+prop_roundtrip_double :: Property
+prop_roundtrip_double =
+  gamble arbitrary $ \(n :: Int64) ->
+  gamble arbitrary $ \(x :: Double) ->
+  let
+    original =
+      0 + x * fromIntegral n
```

Comment:
This `0 +` is not strictly necessary, but it means we get more interesting counterexamples if the code below is changed to `diff_i64 == 0` (otherwise we just get the `0.0` vs `-0.0` case)

tl;dr adding zero to a floating point number normalises -0 => 0